/**
 * Copyright 2020 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

option cc_enable_arenas = true;
import "types.proto";
import "tensor_shape.proto";
package mindspore;

message Tensor {

  // tensor shape info
  TensorShape tensor_shape = 1;

  // tensor content data type
  int32 tensor_type = 2;

  // tensor memory device
  // data located memory device , "DDR" "HBM" OR "NONE"
  string mem_device = 3;
  string name = 4;
  uint64 data_ptr = 5;
  uint64 data_size = 6;
}
